src/VisualStudio/Core/Def/Implementation/ProjectSystem/InvisibleEditor.cs

```diff
@@ -60,14 +62,13 @@ public InvisibleEditor(IServiceProvider serviceProvider, string filePath, IVsHie
                 try
                 {
                     var docData = Marshal.GetObjectForIUnknown(docDataPtr);
-                    _vsTextLines = docData as IVsTextLines;
```

Comment:
It's not clear to me why this was only using an 'as' cast to convert the docData to an IVsTextLines. We had a hard cast to IVsTextBuffer a line later, and although it's _possible_ that somebody could implement IVsTextBuffer but not IVsTextLines, I can't think of any reasonable scenario for that. Worse off, the only objects here that can be passed to GetDocumentBuffer and actually work would be the editor implementations anyways which would implement all the interfaces.

I'm figuring this all just evolved this way for no good reason, so I'm cleaning it up.